examples/tree_1d_dgsem/elixir_shallowwater_multilayer_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,199 @@
+        elseif x[1] <= 70.0
+            b = 0.7
+        elseif x[1] <= 75.0
+            b = 2.2
+        elseif x[1] <= 95.0
+            b = 0.4
+        else
+            b = 1.5
+        end
+
+        # Set zero initial velocity
+        v1_upper = 0.0
+        v1_lower = 0.0
+
+        #= 
+        It is mandatory to shift the water level at dry areas to make sure the water height h
+        stays positive. The system would not be stable for h set to a hard 0 due to division by h in
+        the computation of velocity, e.g., (h v) / h. Therefore, a small dry state threshold
+        with a default value of 5*eps() ≈ 1e-15 in double precision, is set in the constructor above
+        for the ShallowWaterMultiLayerEquations1D and added to the initial condition if h = 0.
+        This default value can be changed within the constructor call depending on the simulation setup.
+        =#
+        H_lower = max(H_lower, b + equations.threshold_limiter)
+        H_upper = max(H_upper, H_lower + equations.threshold_limiter)
+
+        return prim2cons(SVector(H_upper, H_lower, v1_upper, v1_lower, b), equations)
+    end
+end
+
+# Default value for the perturbation
+perturbation = false
+
+initial_condition = initial_condition_twolayer_well_balanced_wet_dry(perturbation,
+                                                                     equations)
+
+###############################################################################
+# Get the DG approximation space
+
+volume_flux = (flux_ersing_etal, flux_nonconservative_ersing_etal)
+surface_flux = (FluxHydrostaticReconstruction(FluxPlusDissipation(flux_ersing_etal,
+                                                                  DissipationLocalLaxFriedrichs()),
+                                              hydrostatic_reconstruction_ersing_etal),
+                FluxHydrostaticReconstruction(flux_nonconservative_ersing_etal,
+                                              hydrostatic_reconstruction_ersing_etal))
+basis = LobattoLegendreBasis(3)
+
+indicator_sc = IndicatorHennemannGassnerShallowWater(equations, basis,
+                                                     alpha_max = 0.5,
+                                                     alpha_min = 0.001,
+                                                     alpha_smooth = true,
+                                                     variable = waterheight_pressure)
+volume_integral = VolumeIntegralShockCapturingHG(indicator_sc;
+                                                 volume_flux_dg = volume_flux,
+                                                 volume_flux_fv = surface_flux)
+solver = DGSEM(basis, surface_flux, volume_integral)
+###############################################################################
+# Get the TreeMesh and setup a periodic mesh
+
+# The domain of interest [0.0, 100.0] is extended towards -28.0 to match the positions of the 
+# disctontinuities with TreeMesh.
```

Comment:
```suggestion
# The domain of interest [0.0, 100.0] is extended towards -28.0 to match the positions of the 
# discontinuities with TreeMesh.
```
Okay, was this because the original initial condition from the Martínez-Aranda et al. paper set the discontinuities at places that the strict refinement strategy of `TreeMesh` cannot reach?

Reply:
Yes, the discontinuities are all located at whole numbers. I extended the domain to length $128=2^7$, such that all discontinuities are located at the interfaces.

Reply:
EZ! That is what I assumed happened.